dap4/d4tests/src/test/java/dap4/test/TestDAP4Client.java

```diff
@@ -0,0 +1,406 @@
+    ClientTest.server = server;
+    ClientTest.servlet = servlet;
+    alltestcases.add(new ClientTest("test_atomic_array.nc"));
+    alltestcases.add(new ClientTest("test_atomic_types.nc"));
+    alltestcases.add(new ClientTest("test_enum.nc"));
+    alltestcases.add(new ClientTest("test_enum_2.nc"));
+    alltestcases.add(new ClientTest("test_enum_array.nc"));
+    alltestcases.add(new ClientTest("test_enum1.nc"));
+    alltestcases.add(new ClientTest("test_fill.nc"));
+    alltestcases.add(new ClientTest("test_groups1.nc"));
+    if (BROKEN)
+      alltestcases.add(new ClientTest("test_misc1.nc")); // 0 size unlimited
+    if (BROKEN)
+      alltestcases.add(new ClientTest("test_one_var.nc")); // 0 size unlimited
+    alltestcases.add(new ClientTest("test_one_vararray.nc"));
+    alltestcases.add(new ClientTest("test_opaque.nc"));
+    alltestcases.add(new ClientTest("test_opaque_array.nc"));
+    alltestcases.add(new ClientTest("test_struct_array.nc"));
+    alltestcases.add(new ClientTest("test_struct_nested.nc"));
+    alltestcases.add(new ClientTest("test_struct_nested3.nc"));
+    alltestcases.add(new ClientTest("test_struct_type.nc"));
+    alltestcases.add(new ClientTest("test_struct1.nc"));
+    alltestcases.add(new ClientTest("test_test.nc"));
+    if (BROKEN)
+      alltestcases.add(new ClientTest("test_unlim.nc")); // ?
+    if (BROKEN)
+      alltestcases.add(new ClientTest("test_unlim1.nc")); // ?
+    if (BROKEN)
+      alltestcases.add(new ClientTest("test_utf8.nc")); // ?
+    alltestcases.add(new ClientTest("test_vlen1.nc"));
+    if (BROKEN)
+      alltestcases.add(new ClientTest("test_vlen2.nc")); // non scalar vlen
+    if (BROKEN)
+      alltestcases.add(new ClientTest("test_vlen3.nc")); // non scalar vlen
+    if (BROKEN)
+      alltestcases.add(new ClientTest("test_vlen4.nc")); // non scalar vlen
+    alltestcases.add(new ClientTest("test_vlen5.nc"));
+    if (BROKEN)
+      alltestcases.add(new ClientTest("test_vlen6.nc")); // non-scalar vlen
+    if (BROKEN)
+      alltestcases.add(new ClientTest("test_vlen7.nc")); // non-scalar vlen
+    if (BROKEN)
+      alltestcases.add(new ClientTest("test_vlen8.nc")); // non-scalar vlen
+    if (BROKEN)
+      alltestcases.add(new ClientTest("test_vlen9.nc")); // non-scalar
+    if (BROKEN)
+      alltestcases.add(new ClientTest("test_vlen10.nc")); // non-scalar
+    if (BROKEN)
+      alltestcases.add(new ClientTest("test_vlen11.nc")); // unknown failure
+    if (BROKEN)
+      alltestcases.add(new ClientTest("test_zerodim.nc")); // non-scalar seq
+    alltestcases.add(new ClientTest("tst_fills.nc"));
+    for (ClientTest test : alltestcases) {
+      if (what == HEADERONLY)
+        test.headeronly();
+    }
+    return true;
+  }
+
+  //////////////////////////////////////////////////
+  // Junit test method
+
+  @Test
+  public void testDAP4Client() throws Exception {
+    boolean pass = true;
+    for (ClientTest testcase : chosentests) {
+      if (!doOneTest(testcase))
```

Comment:
I think at some point we'll want to refactor this to use a Parameterized JUnit test, for better granularity on the test results (and to match the parts of the project that have been updated more recently)